pkg/kapp/resources/identified_resources_pods.go

```diff
@@ -35,31 +37,33 @@ func (w UniquePodWatcher) Watch(podsToWatchCh chan corev1.Pod, cancelCh chan str
 		// Watch Pods in all namespaces first and fallback to the
 		// fallbackAllowedNamespaces if lack of permission
 		namespace := ""
+		var index int
+		var forbiddenNamespaces []string
 		for {
 			podWatcher := NewPodWatcher(
 				w.coreClient.CoreV1().Pods(namespace),
 				metav1.ListOptions{LabelSelector: w.labelSelector.String()},
 			)
-
 			err := podWatcher.Watch(nonUniquePodsToWatchCh, cancelCh)
-			if err == nil {
-				break
-			}
-			if errors.IsForbidden(err) && namespace == "" {
-				// The '-n' flag or default state namespace can specify only 1 namespace, so there
-				// should be at most 1 item in fallbackAllowedNamespaces
-				if len(w.fallbackAllowedNamespaces) > 0 {
-					namespace = w.fallbackAllowedNamespaces[0]
-					if namespace == "" {
+			if err != nil {
+				if namespace != "" {
+					if !errors.IsForbidden(err) {
+						fmt.Printf("Pod watching error: %s\n", err) // TODO
 						break
 					}
+					forbiddenNamespaces = append(forbiddenNamespaces, fmt.Sprintf(`"%s"`, namespace))
 				}
-			} else {
-				fmt.Printf("Pod watching error: %s\n", err) // TODO
+			}
+
+			if index >= len(w.fallbackAllowedNamespaces) {
 				break
 			}
+			namespace = w.fallbackAllowedNamespaces[index]
+			index++
+		}
+		if len(forbiddenNamespaces) > 0 {
+			fmt.Printf(`Pod watching error: pods is forbidden: User cannot list resource "pods" in API group "" in the namespace(s) %s`, strings.Join(forbiddenNamespaces, ", "))
```

Comment:
nit: `pods is forbidden:` looks odd

Reply:
+1

Reply:
I think it is trying to say "pods" as resource rather than as a plural of pod

Fine by me